cmd/zdb/zdb.c

```diff
@@ -2483,6 +2485,10 @@ dump_znode(objset_t *os, uint64_t object, void *data, size_t size)
 		}
 		(void) printf("\tpath	%s\n", path);
 	}
+	if (S_ISLNK(mode) && sa_lookup(hdl, sa_attr_table[ZPL_SYMLINK],
+	    &linktarget, sizeof (uint64_t)) == 0)
```

Comment:
Sorry I didn't catch this in the first review.  But since the `ZPL_SYMLINK` system attribute is variable size you're going to need to use `sa_size` to determine its size and provide that size to `sa_lookup`, instead of `sizeof (uint64_t)`.

Take a look at `dump_znode_sa_xattr()` for a similar example which uses `ZPL_DXATTR` and is also variably sized (although it stored a packed nvlist and not a string).  Adding a similar `static void dump_znode_symlink(sa_handle_t *hdl)` would be a good way to handle this.

Reply:
NP! Factored code into dump_znode_symlink(), tested and pushed. It just returns on error like dump_znode_sa_xattr(), more than happy to make additional changes if necessary. Thanks.